core/src/main/java/org/apache/iceberg/rest/Endpoint.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest;
+
+import java.util.List;
+import org.apache.iceberg.relocated.com.google.common.base.Joiner;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.base.Splitter;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class Endpoint {
+
+  private static final Splitter ENDPOINT_SPLITTER = Splitter.on(" ");
+  private static final Joiner ENDPOINT_JOINER = Joiner.on(" ");
+
+  public abstract String httpVerb();
```

Comment:
In most places, we use `method`.

Reply:
I initially had this as `httpMethod` and renamed it in the latest revision of the code :) I'm fine either way, so renaming this back to `httpMethod`